ussomacaroon/client_test.go

```diff
@@ -0,0 +1,212 @@
+// Copyright 2016 Canonical Ltd.
+// Licensed under the LGPLv3, see LICENCE file for details.
+
+package ussomacaroon_test
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+
+	"github.com/juju/httprequest"
+	jc "github.com/juju/testing/checkers"
+	gc "gopkg.in/check.v1"
+	errgo "gopkg.in/errgo.v1"
+	"gopkg.in/macaroon-bakery.v2-unstable/httpbakery"
+	"gopkg.in/macaroon.v2-unstable"
+
+	"github.com/juju/idmclient/params"
+	"github.com/juju/idmclient/ussomacaroon"
+)
+
+type clientSuite struct {
+	srv      *httptest.Server
+	macaroon *macaroon.Macaroon
+}
+
+var _ = gc.Suite(&clientSuite{})
+
+func (s *clientSuite) SetUpTest(c *gc.C) {
+	s.srv = httptest.NewServer(s)
```

Comment:
Maybe move the ServeHTTP method up here so that is more apparent that this suite is also an handler?